resolver/resolver.cc

```diff
@@ -564,6 +589,9 @@ class ResolveConstantsWalk {
 
     static bool resolveJob(core::Context ctx, ResolutionItem &job) {
         if (isAlreadyResolved(ctx, *job.out)) {
+            if (job.possibleGenericType) {
+                return false;
+            }
```

Comment:
This will have a pretty negative effect on the resolver in the normal case, as generics will never be resolved in the first pass through the tree. What case required unconditionally delaying jobs that look like generics?

Reply:
It's worth noting that this only affects single-package RBI generation mode, since that's the only time we set `possibleGenericType` to true.  (I guess we'll have an extra test-and-branch normally, but that's cheap and branch predictors are pretty good.)

The problem this fixes is that we can wind up in a situation where we have stubbed a constant before we knew that it was a possible generic type.  So when we get a `ResolutionItem` that we know *could* be a generic, we would find that we already stubbed it, and consider this particular job as resolved.  But we needed this particular job to persist until `constantResolutionFailed`, because this job carries information that marks the associated constant as potentially a generic type, and we need to process this job appropriately to make the stub look like a generic type.  Ergo, we couldn't let this job resolve "properly".

I guess this could have possibly bad effects on constants that we *didn't* stub, but look like generics, and therefore we might never consider them resolved.  That does seem bad, so perhaps this check should be a little smarter?

Reply:
Ah, I had missed that we only set `possibleGenericType` when single-package mode is enabled. I'm a lot less worried about performance impact if that's the case.